### PR TITLE
Allow config.Load to load partial config

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/888

### CODE REVIEW

**Comment by pavolloffay** [comment]
Could we have a test that literally loads an empty file and parses it to config object?

**Comment by bogdandrutu** [comment]
@pavolloffay PTAL

File: config/test_helpers.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
 )
 
-// LoadConfigFile loads a config from file.
+// LoadConfigFile loads a config from file, without verifying it.
```
Comment (tigrannajaryan) [inline config/test_helpers.go:27]:
```suggestion
// LoadConfigFile loads a config from file, without validating it.
```

File: config/test_helpers.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ import (
...
 func LoadConfigFile(t *testing.T, fileName string, factories Factories) (*configmodels.Config, error) {
```
Comment (tigrannajaryan) [inline config/test_helpers.go:28]: This changes exported functionality that is used from Contrib repo. I suggest we keep `LoadConfigFile` behavior unchanged (i.e. add `ValidateConfig` call to it).

We probably don't need a version of this function that does not validate the config. I think it is only needed `TestLoadEmptyConfig` so just call `config.Load` directly.

Comment (bogdandrutu) [reply config/test_helpers.go]: Done.

File: config/test_helpers.go (hunk 1 above)
Comment (bogdandrutu) [reply config/test_helpers.go]: No longer a change.

**Pull request merged**